lib/Service/RecentPagesService.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OCA\Collectives\Service;
+
+use OCA\Collectives\Model\PageInfo;
+use OCA\Collectives\Model\RecentPage;
+use OCP\DB\Exception;
+use OCP\DB\QueryBuilder\IQueryBuilder;
+use OCP\Files\IMimeTypeLoader;
+use OCP\IConfig;
+use OCP\IDBConnection;
+use OCP\IURLGenerator;
+use OCP\IUser;
+
+class RecentPagesService {
+
+	public function __construct(
+		protected CollectiveService $collectiveService,
+		protected IDBConnection $dbc,
+		protected IConfig $config,
+		protected IMimeTypeLoader $mimeTypeLoader,
+		protected IURLGenerator $urlGenerator,
+	) {	}
+
+	/**
+	 * @return RecentPage[]
+	 * @throws MissingDependencyException
+	 * @throws Exception
+	 */
+	public function forUser(IUser $user, int $limit = 10): array {
+		try {
+			$collectives = $this->collectiveService->getCollectives($user->getUID());
```

Comment:
I wonder whether it would improve readability to rename `$collectives` to something else (e.g. `$userCollectives`) here, given that `$collectives` is used again for a different purpose down below.